tsl/test/sql/compression_insert.sql.in

```diff
@@ -0,0 +1,550 @@
+INSERT INTO test2(itime ,t , d, charcol)
+values (2, '2', 22, 'thisis22'), (17, '17', 1701, 'thisis1700') ;
+
+COPY test2(itime,t,d,charcol) FROM STDIN DELIMITER ',';
+2,2copy,22,22copy
+17,17copy,1701,1700copy
+\.
+
+SELECT * from test2 where itime = 2 or itime =17
+ORDER BY 1, 2, 3, 4, 5;
+
+DROP TABLE test2;
+
+--- TEST3 tables with defaults ---
+--  sequences, generated values, check constraints into compressed chunks
+CREATE TABLE test2 (timec timestamptz ,
+      i integer CHECK ( i > 10) ,
+      b bigint default 20 ,
+      t text NOT NULL,
+      CONSTRAINT rowconstr CHECK ( b > i )
+);
+
+SELECT table_name from create_hypertable('test2', 'timec', chunk_time_interval=> INTERVAL '7 days');
+
+ALTER TABLE test2 set (timescaledb.compress,
+timescaledb.compress_segmentby = 'b',
+timescaledb.compress_orderby = 'timec DESC');
+
+INSERT INTO test2 values('2020-01-02 11:16:00-05' , 100, 105, 'first' );
+SELECT compress_chunk(c)
+FROM show_chunks('test2') c;
+
+-- test if default value for b is used
+INSERT INTO test2(timec, i, t) values('2020-01-02 10:16:00-05' , 11, 'default' );
+COPY test2(timec,i,t) FROM STDIN DELIMITER ',';
+2020-01-02 10:16:00-05,11,defaultcopy
+\.
+
+SELECT b from test2 ORDER BY 1;
+
+\set ON_ERROR_STOP 0
+--null value for t, should fail
+INSERT INTO test2 values ( '2020-01-02 01:00:00-05', 100, 200, NULL);
+COPY test2 FROM STDIN DELIMITER ',' NULL 'NULL';
+2020-01-02 01:00:00-05,100,200,NULL
+\.
+
+-- i=1, should fail
+INSERT INTO test2 values ( '2020-01-02 01:00:00-05', 1, 10, 'null i');
+COPY test2 FROM STDIN DELIMITER ',';
+2020-01-02 01:00:00-05,1,10,null i
+\.
+-- b < i, should fail
+INSERT INTO test2 values ( '2020-01-02 01:00:00-05', 22, 1, 'null i');
+COPY test2 FROM STDIN DELIMITER ',';
+2020-01-02 01:00:00-05,22,1,null i
+\.
+\set ON_ERROR_STOP 1
+--verify we are still INSERTing into the compressed chunk i.e did not
+--create a new chunk
+SELECT count(c)
+FROM show_chunks('test2') c;
+
+-- TEST4 with sequence .
+CREATE SEQUENCE vessel_id_seq
+    INCREMENT 1
+    START 1    MINVALUE 1
+    MAXVALUE 9223372036854775807
+    CACHE 1;
+
+CREATE TABLE vessels (timec timestamptz ,
+      id bigint NOT NULL DEFAULT nextval('vessel_id_seq'::regclass),
+      i integer CHECK ( i > 10) ,
+      b bigint default 20 ,
+      t text NOT NULL,
+      CONSTRAINT rowconstr CHECK ( b > i )
+);
+
+SELECT table_name from create_hypertable('vessels', 'timec', chunk_time_interval=> INTERVAL '7 days');
+
+ALTER TABLE vessels set (timescaledb.compress,
+timescaledb.compress_segmentby = 'b',
+timescaledb.compress_orderby = 'timec DESC');
+
+INSERT INTO vessels(timec, i, b, t) values('2020-01-02 11:16:00-05' , 100, 105, 'first' );
+SELECT compress_chunk(c)
+FROM show_chunks('vessels') c;
+
+-- test if default value for b and sequence value for id is used
+INSERT INTO vessels(timec, i, t) values('2020-01-02 10:16:00-05' , 11, 'default' );
+COPY vessels(timec,i,t )FROM STDIN DELIMITER ',';
```

Comment:
Do we also have a test with COPY directly into the chunk when it is compressed?

Reply:
No. It needs to go through chunk_insert_state to get directed to the correct chunk.